src/modules/producer.js

```diff
@@ -1,175 +1,200 @@
-var utils = require('./utils'),
-  uuid = require('node-uuid'),
-  parsers = require('./message-parsers'),
-  Deferred = require('../classes/deferred');
-
-var amqpRPCQueues = {};
+const utils = require('./utils');
+const uuid = require('node-uuid');
+const parsers = require('./message-parsers');
+const Deferred = require('../classes/deferred');
+const assert = require('assert');
 
 const ERRORS = {
   TIMEOUT: 'Timeout reached',
   BUFFER_FULL: 'Buffer is full'
 };
 
-/**
- * Create a RPC-ready queue
- * @param  {string} queue the queue name in which we send a RPC request
- * @return {Promise}       Resolves when answer response queue is ready to receive messages
- */
-function createRpcQueue(queue) {
-  if (!amqpRPCQueues[queue]) {
-    amqpRPCQueues[queue] = {};
+class Producer {
+  constructor(connection) {
+    this.amqpRPCQueues = {};
+    this._conn = connection;
+    this.channel = null;
   }
 
-  let rpcQueue = amqpRPCQueues[queue];
-  if (rpcQueue.queue) return Promise.resolve(rpcQueue.queue);
+  get conn() {
```

Comment:
These two are used all over the project as properties. This is just an es6 feature to simulate property via get set functions. Should I still rename it?